README.md

```diff
@@ -318,64 +318,47 @@ var output = translate.prettyPrintLegacyAssemblyJSON(assemblyJSON, sourceCode)
 
 ## Browser Usage
 
-Add the version of `solc` you want to use into `index.html`:
+We currently compile the wasm module in `soljson.js` synchronously (see [here](https://github.com/ethereum/solc-js/issues/581) for more information), which may not work in Chrome-based
+browsers since it limits the size of modules that can be compiled/fetched synchronously to avoid blocking the main thread.
+Thus, the recommended way to use `solc` in a web browser is through a [web worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers).
```

Comment:
I think we can be even more aggressive with this.
I.e. something along the lines of ``Compilation is generally a long-running and resource intensive task that cannot reasonably be performed in the main thread. The only supported way to use `solc` in a web browser is through a [web worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers).``

I actually think browsers would even complain about asynchronously running the compiler due to it running too long and the browser suspecting it hangs - but I'm not 100% sure that's true :-). Web workers are really the proper way to do this in general, aren't they?

Reply:
Firefox does not complain about it, at least not on the simple tests that I did. But chromium immediately complains emitting an warning that `WebAssembly.Compile` is disallowed on the main thread if the buffer size is larger than 4KB.

In general I do agree that web workers are the proper way to do this, although some web browsers will still allow you to compile web assembly in the browser's main thread.

Reply:
I know - but what's the use case for anything that works in *some*, but not all browsers :-)? Resp. "supported use" would have to mean something that works universally. I.e. as long as something not works in *all* browsers, I don't think it makes sense to suggest it in any way.